src/components/modals/WcSmartContractModal.tsx

```diff
@@ -122,6 +123,7 @@ export const WcSmartContractModal = (props: Props) => {
           const tx = await wallet.makeSpend(spendInfo)
           const signedTx = await wallet.signTx(tx)
           const sentTx = await wallet.broadcastTx(signedTx)
+          await wallet.saveTx(sentTx)
```

Comment:
Can we put this below `walletConnect.approveRequest`? If WalletConnect fails, we don't want to call `saveTx` at all.